File: fe-admin/src/app/core/model/response/procedure-response.ts
import {ProcedureCategoryResponse} from "./procedure-category-response";

export interface ProcedureResponse{
    id:number;
    code:string;
    name:string;
    description:string;
    price:number;
    unit:string;
    durationMinutes:number;
    isActive:boolean;
    version:number;
    procedureCategory:ProcedureCategoryResponse;
    createdAt:Date;
    modifiedAt:Date;
}

File: be-api/src/main/java/cd/beapi/dto/response/ProcedureResponse.java
package cd.beapi.dto.response;

import java.math.BigDecimal;
import java.time.Instant;

public record ProcedureResponse(
        Long id,
        String code,
        String name,
        String description,
        BigDecimal price,
        String unit,
        Integer durationMinutes,
        Boolean isActive,
        Long version,
        ProcedureCategoryResponse procedureCategory,
        Instant createdAt,
        Instant modifiedAt) {
}
